docs/root/version_history/current.rst

```diff
@@ -90,6 +90,7 @@ New Features
 * router: added new :ref:`host_rewrite_path_regex <envoy_v3_api_field_config.route.v3.RouteAction.host_rewrite_path_regex>`
   option, which allows rewriting Host header based on path.
 * router: added support for DYNAMIC_METADATA :ref:`header formatter <config_http_conn_man_headers_custom_request_headers>`.
+* router_check_tool: added support for `request_header_matches`, `response_header_matches`.
```

Comment:
doc nit: please ref link to the router check tool page or to the new fields with an anchor.